components/src/preact/mutationFilter/mutation-filter.tsx

```diff
@@ -84,170 +93,160 @@ export const MutationFilterInner: FunctionComponent<MutationFilterInnerProps> =
-
-    const selectorRef = useRef<HTMLDivElement>(null);
+    const handleSelectedItemsChanged = (newSelectedItems: MutationFilterItem[]) => {
+        fireChangeEvent(newSelectedItems);
+        setSelectedItems(newSelectedItems);
+    };
 
-    const handleInputChange = (newValue: string) => {
-        if (newValue.includes(',') || newValue.includes(';')) {
-            handleCommaSeparatedInput(newValue);
-        } else {
-            setInputValue(newValue);
-            const result = parseAndValidateMutation(newValue, referenceGenome);
-            setOption(result);
+    const handleNewSelectedItem = (selectedItem: MutationFilterItem | null | undefined) => {
+        if (selectedItem) {
+            handleSelectedItemsChanged([...selectedItems, selectedItem]);
+            setInputValue('');
+            setItemCandidate(null);
+            setShowErrorIndicator(false);
         }
     };
 
-    const handleCommaSeparatedInput = (inputValue: string) => {
-        const inputValues = inputValue.split(/[,;]/);
-        let newSelectedOptions = selectedFilters;
-        let updated: boolean = false;
-        const invalidQueries: string[] = [];
-        for (const value of inputValues) {
-            const trimmedValue = value.trim();
-            const parsedMutation = parseAndValidateMutation(trimmedValue, referenceGenome);
-            if (parsedMutation) {
-                const type = parsedMutation.type;
-                if (!selectedFilters[type].some((i) => parsedMutation.value.toString() === i.toString())) {
-                    newSelectedOptions = {
-                        ...newSelectedOptions,
-                        [parsedMutation.type]: [...newSelectedOptions[parsedMutation.type], parsedMutation.value],
-                    };
-                    updated = true;
+    const handleInputChange = (newInputValue: string | undefined) => {
+        setShowErrorIndicator(false);
+        if (newInputValue?.includes(',')) {
+            const values = newInputValue?.split(',').map((value) => {
+                return { value, parsedValue: parseAndValidateMutation(value.trim(), referenceGenome) };
+            });
+            const validEntries = values.map((value) => value.parsedValue).filter((value) => value !== null);
+            const invalidInput = values
+                .filter((value) => value.parsedValue === null)
+                .map((value) => value.value.trim())
+                .join(',');
+
+            handleSelectedItemsChanged([...selectedItems, ...validEntries]);
+            setInputValue(invalidInput);
+            setItemCandidate(null);
+        } else {
+            setInputValue(newInputValue ?? '');
+            if (newInputValue !== undefined) {
+                const candidate = parseAndValidateMutation(newInputValue, referenceGenome);
+                if (candidate) {
+                    setItemCandidate(candidate);
+                } else {
+                    setItemCandidate(null);
```

Comment:
`candidate` is already a mutation or null.
```suggestion
                    setItemCandidate(candidate);
```